diff --git a/splituapp b/splituapp
index 587543b..69abce1 100755
--- a/splituapp
+++ b/splituapp
@@ -19,8 +19,8 @@ import sys
 import struct
 import argparse
 
-def extract(source, flist):
-    outdir = 'output'
+def extract(source, output, flist):
+    outdir = output
     img_files = set()
 
     try:
@@ -154,6 +154,11 @@ if __name__ == '__main__':
     optional = parser.add_argument_group('Optional')
     optional.add_argument("-h", "--help", action="help", help="show this help message and exit")
     optional.add_argument("-l", "--list", nargs="*", metavar=('img1', 'img2'), help="List of img files to extract")
+    optional.add_argument("-o", "--output", action="store", help="Put output in the given folder")
     args = parser.parse_args()
 
-    extract(args.filename, args.list)
+    output_path = "output"
+    if (args.output != None):
+        output = args.output
+
+    extract(args.filename, output, args.list)
