diff --git a/src/dex_instruction.h b/src/dex_instruction.h
index 204d6e6..022513a 100644
--- a/src/dex_instruction.h
+++ b/src/dex_instruction.h
@@ -275,7 +275,7 @@ s4 dexInstr_getVRegH(u2 *);
 u2 dexInstr_getVRegH_45cc(u2 *);
 u2 dexInstr_getVRegH_4rcc(u2 *);
 bool dexInstr_hasVarArgs(u2 *);
-void dexInstr_getVarArgs(u2 *, u4[]);
+void dexInstr_getVarArgs(u2 *, u4[kMaxVarArgRegs]);
 
 // Set register functions
 void dexInstr_SetVRegA_10x(u2 *, u1);
diff --git a/src/utils.c b/src/utils.c
index 0269f37..439c3a0 100644
--- a/src/utils.c
+++ b/src/utils.c
@@ -376,7 +376,7 @@ fini:
 }
 
 char *utils_fileBasename(char const *path) {
-  char *s = strrchr(path, '/');
+  const char *s = strrchr(path, '/');
   if (!s) {
     return strdup(path);
   } else {
